connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java

```diff
@@ -61,4 +68,23 @@ public void testLoadingPluginClass() throws ClassNotFoundException {
             assertNotNull(classLoader.pluginClassLoader(pluginClassName));
         }
     }
+
```

Comment:
Nice to have test cases. Are all the cases that we want to cover? Did also fail with an empty directory, or a directory that contained an unparsable file (not a `jar`, but let's say a text file). If so, it'd be nice to have tests for these edge cases too. In general, as inspiration we could use how `/usr/share/java` looks on typical linux installations.

Reply:
I have added more testcases to cover these scenarios.